Mkdocs material documentation (#58)
slub/ocrd_kitodo: docs/setup/configure-external.md

> This page describes the configuration of an external OCR-D Controller and Kitodo.Production instance.
> If do **not** wish to run the built-in OCR-D Controller and/or Kitodo.Production module read the appropriate sections.
> Otherwise you can skip this page and continue with [Preparation](preparation.md).

Reviewer: ```suggestion
This page describes how to configure an **external** OCR-D Controller or Kitodo.Production instance.
So if you do **not** wish to run the **built-in** OCR-D Controller and/or Kitodo.Production module, 
then read the respective section below.
Otherwise, you can skip this page and continue with [Preparation](preparation.md).
```

- `git clone https://github.com/slub/ocrd_controller && cd ocrd_controller` (otherwise)

#### credentials

You will also need the public SSH key file for login from the Manager to the Controller,
created on the ocrd_kitodo host via `make prepare-keys` under `ocrd/controller/.ssh/authorized_keys`
ready on this machine.

### Setup

There are multiple scenarios:

#### Docker, running manually via make

Following the [standalone documentation](https://github.com/slub/ocrd_controller#starting-and-mounting),
choose your parameter values (persistent storage, key, CPUs, SSH port) and **make run**.

For example:

    export DATA=/mnt/workspaces
    export MODELS=~/.local/share
    export CONFIG=~/.config/share
    export KEYS=authorized_keys
    export WORKERS=2
    export PORT=8022
    make run

#### Docker, running manually via Docker Compose

Mimicking the Docker Compose / `.env` approach from the top-level repo,
choose your parameter values (persistent storage, key, CPUs, SSH port) and **docker compose up**.

For example:

    echo "CONTROLLER_IMAGE=ghcr.io/slub/ocrd_controller:latest" > .env

>     echo "CONTROLLER_HOST=ocrd-server" >> .env

Reviewer: ```suggestion
    echo "CONTROLLER_HOST=ocrd-server" >> .env
    echo "CONTROLLER_PORT_SSH=8022" >> .env
```
